[bug][CMakeConfigDeps] cpp_info flags need concatenation  with ';'

### Describe the bug

Hi,

While using CMakeConfigDeps (instead of CMakeDeps) I stumbled over a linker error in Windows. I guess the problem comes from the fact that CmakeConfigDeps concatenates several flags with spaces instead of ';'.

```
conanfile.py:
========
def package_info(self):
    self.cpp_info.components["comp"].sharedlinkflags += ["/INCLUDE:symbol1"]
    self.cpp_info.components["comp"].sharedlinkflags += ["/INCLUDE:symbol2"]

CMakeDeps generates (works):
===============
set(lib_lib_comp_SHARED_LINK_FLAGS_DEBUG "/INCLUDE:symbol1;/INCLUDE:symbol2")

CMakeConfigDeps generates (error):
========================
set_property(TARGET lib::comp APPEND PROPERTY INTERFACE_LINK_OPTIONS
            "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:$<$<CONFIG:DEBUG>:/INCLUDE:symbol1 /INCLUDE:symbol2"

Which leads to:
LINK : error LNK2001: unresolved external symbol symbol1 /INCLUDE:symbol2
```
When I swap the space in the (generated CMake config files) to ';' it works again.
```
set_property(TARGET lib::comp APPEND PROPERTY INTERFACE_LINK_OPTIONS
            "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:$<$<CONFIG:DEBUG>:/INCLUDE:symbol1;/INCLUDE:symbol2"
```
Best,
Michael

### How to reproduce it

_No response_

--- a/conan/tools/cmake/cmakedeps2/target_configuration.py
+++ b/conan/tools/cmake/cmakedeps2/target_configuration.py
@@ -345,13 +345,13 @@ class TargetConfigurationTemplate2:
         {% if lib_info.get("sharedlinkflags") %}
         {% set linkflags = config_wrapper(config, lib_info["sharedlinkflags"]) %}
         set_property(TARGET {{lib}} APPEND PROPERTY INTERFACE_LINK_OPTIONS
-                     "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:{{linkflags}}>"
-                     "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:{{linkflags}}>")
+                     $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:{{linkflags}}>
+                     $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:{{linkflags}}>)
         {% endif %}
         {% if lib_info.get("exelinkflags") %}
         {% set exeflags = config_wrapper(config, lib_info["exelinkflags"]) %}
         set_property(TARGET {{lib}} APPEND PROPERTY INTERFACE_LINK_OPTIONS
-                     "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:{{exeflags}}>")
+                     $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:{{exeflags}}>)
         {% endif %}
 
         {% if lib_info.get("link_languages") %}

--- a/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
+++ b/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
@@ -842,8 +842,8 @@ class TestHeaders:
                         self.cpp_info.system_libs = ["m", "dl"]
                         # Just to verify CMake don't break
                     if self.settings.compiler == "gcc":
-                        self.cpp_info.sharedlinkflags = ["-z now", "-z relro"]
-                        self.cpp_info.exelinkflags = ["-z now", "-z relro"]
+                        self.cpp_info.sharedlinkflags = ["-z lazy", "-s"]
+                        self.cpp_info.exelinkflags = ["-z lazy", "-s"]
              """)
         engine_h = textwrap.dedent("""
             #pragma once

